data/transactions/logic/eval.go

```diff
@@ -4801,19 +4807,49 @@ func opItxnSubmit(cx *EvalContext) error {
 	}
 
 	ep := NewInnerEvalParams(cx.subtxns, cx)
+
+	if ep.Debugger != nil {
+		err := ep.Debugger.BeforeInnerTxnGroup(ep)
+		if err != nil {
+			return fmt.Errorf("error while running debugger BeforeInnerTxnGroup hook: %w", err)
+		}
+	}
+
 	for i := range ep.TxnGroup {
+		if ep.Debugger != nil {
+			err := ep.Debugger.BeforeTxn(ep, i)
```

Comment:
Do you have reasons in mind why a debugger is allowed to error? They seem purely observational.

Reply:
I've removed the error return types from tracer hooks. Now if a tracer needs to error, it can do one of the following:
* Panic
* Log the error
* Store the error as a local struct field. This field can be inspected by the caller after evaluation has finished. This way, the caller can handle tracer errors separately from txn/program eval errors.